### PR TITLE
Update ogr.py to enable CURVEPOLYGON

### PR DESCRIPTION
**Pull request by reinij**
Allow to use CURVEPOLYGON from WFS forcing geometry to POLYGON to enable coversion to geoJSON.

### CODE REVIEW

File: pygeoapi/provider/ogr.py [hunk 1]
```diff
@@ -457,10 +457,16 @@ def _get_next_feature(self, layer, feature_id):
...
+        #convert curvepolygon to polygon as GeoJSON doesn't support curves
+        if geom.GetGeometryType() == osgeo_ogr.wkbCurvePolygon:
+            geom_polygon = osgeo_ogr.ForceToPolygon(geom)
+            ogr_feature.SetGeometry(geom_polygon)
+        
         if self.transform_out:
             # Optionally reproject the geometry
             geom.Transform(self.transform_out)
-
+
```
Comment (tomkralidis) [inline pygeoapi/provider/ogr.py:469]: This should be removed as it will result in flake8 errors.

File: pygeoapi/provider/ogr.py [hunk 2]
```diff
@@ -457,10 +457,16 @@ def _get_next_feature(self, layer, feature_id):
 
     def _ogr_feature_to_json(self, ogr_feature):
         geom = ogr_feature.GetGeometryRef()
+        
+        #convert curvepolygon to polygon as GeoJSON doesn't support curves
+        if geom.GetGeometryType() == osgeo_ogr.wkbCurvePolygon:
```
Comment (tomkralidis) [inline pygeoapi/provider/ogr.py:462]: Curious, do we need to consider other geometry types that may need to be forced into polygon geometries?

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: pygeoapi/provider/ogr.py (hunk 2 above)
Comment (justb4) [reply pygeoapi/provider/ogr.py:462]: Yes, complicated. Hence my question in #861: *I see that there is a whole bunch:
https://gdal.org/development/rfc/rfc49_curve_geometries.html* . I  am no expert on this but:

* is this PR solving only a single use-case (single non-linear geometry)?
* can this be generically solved via an OGR option setting?

**Comment by francbartoli** [comment]
Hey @reinij, what is the status of this PR? Do you like to resolve the comments or get help from us?

**Comment by pvgenuchten** [comment]
interesting topic, also consider that after some discussion on the feat-geom group there is now a proposal to add special geometry types to feat-geom, see https://github.com/opengeospatial/ogc-feat-geo-json/blob/main/proposals/circular-geometry-objects.adoc

**Comment by tomkralidis** [comment]
@reinij can we get a status on this PR?

**Comment by reinij** [comment]
I currently have the feeling that this should be canceled and let it be as it is now, i.e. the geometries are not automatically converted but it is assumed that the GML is a valid GeoJSON geometry.

The future JSON-FG then offers all the necessary geometries.

**Pull request closed without merging**